conformance/system_macros/delta.ion

```diff
@@ -0,0 +1,12 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Test Cases:
+// delta can be invoked by name, address, qualified address, qualified name, binary address, and binary system e-expression op code.
+// the arguments of delta must be non-null integers
+// the "initial" argument is required
+// the "delta" argument can be zero or more values
+// any annotations on the arguments are silently dropped
```

Comment:
I don't have a strong preference one way or another, but I think we should either forbid or ignore annotations consistently across the system macros.

EDIT: Skimming ahead, it looks like we mostly drop them. I think I saw a "forbid" case earlier, which may have just been in `add_macros` due to constraints in the `module` clause syntax. If that's the case, we're good.